packages/flutter_tools/test/commands.shard/hermetic/run_test.dart

```diff
@@ -125,6 +125,55 @@ void main() {
       DeviceManager: () => MockDeviceManager(),
     });
 
+    testUsingContext('Walks upward looking for a pubspec.yaml and succeeds if found', () async {
+      globals.fs.file('pubspec.yaml').createSync();
+      globals.fs.file('.packages')
+        ..createSync()
+        ..writeAsStringSync('Not a valid package');
+
+      globals.fs.currentDirectory = globals.fs.directory(globals.fs.path.join('a', 'b', 'c'))
+        ..createSync(recursive: true);
+
+      final RunCommand command = RunCommand();
+      applyMocksToCommand(command);
+      try {
+        await createTestCommandRunner(command).run(<String>[
+          'run',
+          '--fast-start',
+          '--no-pub',
+        ]);
+        fail('Expect exception');
+      } catch (e) {
+        expect(e, isInstanceOf<ToolExit>());
+        expect(e.toString(), contains('Target file "lib/main.dart" not found.'));
```

Comment:
should we also expect to find the log told us what directory we ended up in?  Or am I misunderstanding what this test is actually setting up?